libs/pageserver_api/src/key.rs

```diff
@@ -569,6 +573,37 @@ pub const AUX_FILES_KEY: Key = Key {
     field6: 2,
 };
 
+#[inline(always)]
+pub fn repl_origin_key(origin_id: RepOriginId) -> Key {
+    Key {
+        field1: REPL_ORIGIN_KEY_PREFIX,
+        field2: 0,
+        field3: 0,
+        field4: 0,
+        field5: 0,
+        field6: origin_id as u32,
+    }
+}
+
+/// Get the range of replorigin keys.
+pub fn repl_origin_key_range() -> Range<Key> {
+    Key {
+        field1: REPL_ORIGIN_KEY_PREFIX,
+        field2: 0,
+        field3: 0,
+        field4: 0,
+        field5: 0,
+        field6: 0,
+    }..Key {
+        field1: REPL_ORIGIN_KEY_PREFIX,
+        field2: 0,
+        field3: 0,
+        field4: 0,
+        field5: 0,
+        field6: 0x10000,
```

Comment:
Can you use RepOriginId::MAX here?  (if the FFI declares it as a u16)

Or some other way make it clear here that 0x10000 is used because RepOriginId is a 16 bit type: otherwise it looks like a magic number.

Reply:
It is exclusive boundary. So we can write it as `RepOrigin::MAX::uint32+1` but IMHO 0x10000 looks much more readable:)
In any case, my intention was to make range as small as possible, because I saw code pin PS which iterates through all keys in their range. But @VladLazar  said that it is not used in this case. So may be it is better to keep original range with `field1: REPL_ORIGIN_KEY_PREFIX+1`

Reply:
> because I saw code pin PS which iterates through all keys in their range

Just to confirm. `Timeline::scan` doesn't do that

Reply:
I'm fine with a smaller range right now. The storage engine does not iterate all keys for sparse keyspaces.